pkg/cloudprovider/aws/suite_test.go

```diff
@@ -370,6 +370,50 @@ var _ = Describe("Allocation", func() {
 					Expect(resources.Pods().Value()).ToNot(BeNumerically("==", 110))
 				}
 			})
+			Context("AL2", func() {
+				It("should calculate memory overhead based on eni limited pods when ENI limited", func() {
+					opts.AWSENILimitedPodDensity = true
+					opts.VMMemoryOverhead = 0 // cutting a factor out of the equation
+					provider.AMIFamily = &v1alpha1.AMIFamilyAL2
+					instanceInfo, err := instanceTypeProvider.getInstanceTypes(ctx, provider)
+					Expect(err).To(BeNil())
+					it := NewInstanceType(injection.WithOptions(ctx, opts), instanceInfo["m5.xlarge"], 0, provider, nil)
+					overhead := it.Overhead()
+					Expect(overhead.Memory().String()).To(Equal("3073Mi"))
```

Comment:
I was very confused reading `3073Mi` tbh, because that couldn't be right for an m5.xlarge. 
Did a little debugging, and the problem is in our [mocked EC2 response here](https://github.com/aws/karpenter/blob/main/pkg/cloudprovider/aws/fake/ec2api.go#L375). That should be `15` and not `60` which gives you an actual memory overhead of `1093Mi` instead for an m5.xlarge. 

Would you mind switching that 60 to a 15 and updating this PR?

Reply:
Good catch! I've pushed a commit that updates the mock response and corresponding assertions. I did manually calculate the values to verify during the implementation, but honestly never thought about the logic of the two values. This should make the intention much more clear.